tests/mocks/mock_pool.go

```diff
@@ -0,0 +1,684 @@
+// Code generated by MockGen. DO NOT EDIT.
```

Comment:
Oh interesting, is there a specific command to generate mock file for a specific file?

Reply:
Yes! The [mockgen CLI](https://github.com/golang/mock#running-mockgen) handles automatic generation of mock interfaces.